crates/sui-framework/packages/sui-framework/sources/coin.move

```diff
@@ -54,6 +56,19 @@ module sui::coin {
         total_supply: Supply<T>
     }
 
+    struct FreezeCap<phantom T> has key, store {
```

Comment:
I think we might want to put this logic in its own module + make `coin` a friend. If we want to allow the freezing functionality for other framework types, or for third-party types in the future, I think this will end up being cleaner.